python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java

```diff
@@ -141,8 +145,44 @@ private FunctionType buildFunctionType(FunctionDef functionDef) {
 
   @Override
   public void visitImportName(ImportName importName) {
-    //createImportedNames(importName.modules(), null, Collections.emptyList());
-    super.visitImportName(importName);
+    importName.modules()
+      .forEach(aliasedName -> {
+        var names = aliasedName.dottedName().names();
+        var fqn = names
+              .stream().map(Name::name)
+              .toList();
+        var module = projectLevelTypeTable.getModule(fqn);
+        setTypeToName(aliasedName.alias(), module);
+
+        for (int i = names.size() - 1; i >= 0; i--) {
+          setTypeToName(names.get(i), module);
+          module = Optional.ofNullable(module)
+            .map(ModuleType::parent)
+            .orElse(null);
+        }
+      });
+  }
+
+  @Override
+  public void visitImportFrom(ImportFrom importFrom) {
+    var names = importFrom.module().names();
+    var fqn = names
+      .stream().map(Name::name)
+      .toList();
+
+    var module = projectLevelTypeTable.getModule(fqn);
+    importFrom.importedNames().forEach(aliasedName -> aliasedName
+      .dottedName()
+      .names()
+      .stream()
+      .findFirst()
+      .ifPresent(name -> {
+        var type = module.resolveMember(name.name());
+        setTypeToName(name, type);
```

Comment:
If the name actually has an alias, is the type also bound the the original name?
If not we should probably avoid conflicts here (in case the name has an alias specifically not to clash with the same name already defined in the namespace).

Reply:
tests added